src/Features/Core/Portable/TodoComments/AbstractTodoCommentService.cs

```diff
@@ -3,16 +3,20 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Remote;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.TodoComments
 {
     internal abstract class AbstractTodoCommentService : ITodoCommentService
     {
+        private readonly ConditionalWeakTable<RemoteHostClient, KeepAliveSession> _sessions = new ConditionalWeakTable<RemoteHostClient, KeepAliveSession>();
```

Comment:
client is workspace service, and this service can get any arbitary document from workspace.